framework/filters/PageCache.php

```diff
@@ -323,12 +323,6 @@ protected function calculateCacheKey()
         if ($this->varyByRoute) {
             $key[] = Yii::$app->requestedRoute;
         }
-        if (is_array($this->variations)) {
-            foreach ($this->variations as $value) {
-                $key[] = $value;
-            }
-        }
-
-        return $key;
+        return array_merge($key, (array)$this->variations);
```

Comment:
These are not equivalent. `array_merge` keeps keys while `foreach` that was there does not.

Reply:
Are you sure it's OK w/ keys?

Reply:
It should be ok. [it is used as a key for cache](https://github.com/yiisoft/yii2/blob/b0aa7a73de8f972cb4e02e6ae6d3896b0f1a9a11/framework/caching/CacheInterface.php#L97)
> $key a key identifying the value to be cached. This can be a simple string or a complex data structure consisting of factors representing the key
 
If someone introduced array with keys for `variation` it should IMO affect as a factor